### PR TITLE
Add setup method

### PR DESCRIPTION
**Pull request by shohi**

### CODE REVIEW

File: lua/auto-session.lua [hunk 1]
```diff
@@ -62,11 +64,28 @@ end
...
+end
+
+local AutoSession = {
+  conf = {}
+}
 
 local function getSessionNameFromCwd()
   local cwd = vim.fn.getcwd()
-  return cwd:gsub("/", "-")
+  return cwd:gsub("/", "_")
```
Comment (rmagatti) [inline lua/auto-session.lua:88]: I wonder if there's an easy way to make this not break currently existing sessions using `_`.
I don't particularly mind using a dash or an underscore, but I wouldn't want to break people's existing sessions.

**Review by rmagatti — CHANGES_REQUESTED** [review]
Hey first of all, thanks for the pull request!

This looks mostly good to me.
The one downside is it breaks currently existing sessions using a `-`.

Maybe there's something we can improve there?

File: lua/auto-session.lua (hunk 1 above)
Comment (rmagatti) [reply lua/auto-session.lua:88]: On second thought I think neither dash nor underscore is the ideal character to use.
Probably better to go with the same thing Vim already uses for undo files, which is `%`.

Files would be `%Users%username%path%of%session.vim`

**Comment by rmagatti** [comment]
I just pushed an updated to address this change to `%` while still handling legacy files.
After fixing conflicts and leaving out the to underscore change, this PR should be good to go!

Thanks again for submitting it!

File: lua/auto-session.lua (hunk 1 above)
Comment (shohi) [reply lua/auto-session.lua:88]: Yeah, I agree with you, `%` sounds better. The reason for using underscore over dash is that leading dash is not very friendly for shell commands, e.g. deleting such files needs escaping.

**Comment by shohi** [comment]
Sure, I'll update the PR with your latest change. Thanks for the plugin!

**Review by rmagatti — APPROVED** [review]

File: lua/auto-session.lua (hunk 1 above)
Comment (rmagatti) [reply lua/auto-session.lua]: Yeah I realized the issues with `-` too late in the game 😅.
This looks great now, merging!

🙇

**Pull request merged**